### PR TITLE
Make service e2e tests run in timestamped namespaces.

### PR DESCRIPTION
**Pull request by ghost**
Make some service e2e tests run in timestamped namespaces to prevent name collisions across multiple runs.

Shortly I'll migrate all of the tests to their own namespaces, but right now some of the break if they don't run in api.NamespaceDefault (i.e. "default").  I'll fix that in a separate PR.

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -34,11 +34,15 @@ import (
...
 var _ = Describe("Services", func() {
 	var c *client.Client
+	// Use these in tests.  They're unique for each test to prevent name collisions.
+	var namespace0, namespace1 string
 
 	BeforeEach(func() {
 		var err error
 		c, err = loadClient()
 		Expect(err).NotTo(HaveOccurred())
+		namespace0 = "e2e-ns-" + dateStamp() + "-0"
```
Comment (ixdy) [inline test/e2e/service.go]: maybe include 'services' in the name, so we can tell where this comes from? or 'service.go' or something like that.

File: test/e2e/service.go [hunk 2]
```diff
@@ -455,3 +447,10 @@ func addEndpointPodOrFail(c *client.Client, ns, name string, labels map[string]s
 	_, err := c.Pods(ns).Create(pod)
 	Expect(err).NotTo(HaveOccurred())
 }
+
+// dateStamp returns the current time as a string "YYYY-MM-DDTHHMMSS"
+// Handy for unique names across test runs
+func dateStamp() string {
```
Comment (ixdy) [inline test/e2e/service.go]: wondering if instead we should have a function in test/e2e/util.go which generates a namespace name (perhaps parameterized on test name etc as well)?

File: test/e2e/service.go [hunk 3]
```diff
@@ -266,10 +270,6 @@ var _ = Describe("Services", func() {
 				CreateExternalLoadBalancer: true,
 			},
 		}
-
-		By("cleaning up previous service " + serviceName + " from namespace " + ns)
```
Comment (ixdy) [inline test/e2e/service.go]: why was this removed?

File: test/e2e/service.go (hunk 1 above)
Comment (ixdy) [inline test/e2e/service.go]: also, should there be a step at some point which deletes everything from these namespaces?

Comment (quinton-hoole) [inline test/e2e/service.go]: I had to shorten then name to prevent the concatenation of namespace and name from exceeding the 61 character limit imposed by GCE.  That's essentially a bug in Kubernetes, that will have to be fixed separately.  Kubernetes just concatenates them, and fils if the result exceeds the limit of the cloud provider.  I suggest that we leave as is until that bug is fixed.

Comment (quinton-hoole) [inline test/e2e/service.go]: As for deleting the namespaces, yes, I will add some additional cleanup code in AfterEach(), but that requires a PR that derekwaynecarr@ is still working on (I believe), to allow me to delete entire namespaces.

File: test/e2e/service.go (hunk 2 above)
Comment (quinton-hoole) [inline test/e2e/service.go]: Agreed.  I decided to keep everything inside this test for now, just to verify that it stabilizes Jenkins e2e over the weekend.  I plan to clean all of this up nicely across all e2e tests soon, but that is a bigger and lower priority item.

File: test/e2e/service.go (hunk 3 above)
Comment (quinton-hoole) [inline test/e2e/service.go]: It doesn't really achieve much in the normal Jenkins case, as in the new K8S cluster, the old service doesn't exist anyway.  It's only useful if you run the same test twice against the same Kubernetes cluster instance, which almost never happens.  And I'm about to but this test in its own namespace anyway, so even that case won't work soon.

Comment (ixdy) [inline test/e2e/service.go]: I think that some developers start a k8s cluster and run tests multiple times.

We shouldn't remove the cleanup until the namespaces are unique.

Comment (quinton-hoole) [inline test/e2e/service.go]: Fair enough.  I'll put it back.

Comment (quinton-hoole) [inline test/e2e/service.go]: Done in commit 24f4561

Comment (derekwaynecarr) [inline test/e2e/service.go]: FYI - Namespace deletion was merged

Comment (quinton-hoole) [inline test/e2e/service.go]: Very cool. Thanks Derek. I'll take your new code for a spin next week.

File: test/e2e/service.go (hunk 1 above)
Comment (quinton-hoole) [inline test/e2e/service.go]: PS: I suggest merging as is for now, and adding namespace deletion in a second pass.  What we have here already is a vast improvement over the status quo.  Adding namespace deletion will be a further improvement.

**Comment by ixdy** [comment]
LGTM - but can you please squash the two commits?

**Comment by quinton-hoole** [comment]
Done (commit a4dde6).  Thanks.  If you could merge before the end of the day that would be great.
